jenkins_pipelines/environments/common/pipeline-build-validation-cleanup.groovy

```diff
@@ -0,0 +1,249 @@
+                """
+
+            }
+
+            stage("Extract the controller and server hostname") {
+                try {
+                    controllerHostname = sh(
+                            script: """
+                            set -e
+                            cd ${localSumaformDirPath}
+                            terraform output -json configuration | jq -r '.controller.hostname'
+                        """,
+                            returnStdout: true
+                    ).trim()
+
+                    serverHostname = sh(
+                            script: """
+                            set -e
+                            cd ${localSumaformDirPath}
+                            terraform output -json configuration | jq -r '.server.hostname'
+                        """,
+                            returnStdout: true
+                    ).trim()
+
+                    // Print the values for confirmation
+                    echo "Extracted controller hostname: ${controllerHostname}"
+                    echo "Extracted server hostname: ${serverHostname}"
+
+                } catch (Exception e) {
+                    error("Failed to extract hostnames: ${e.message}")
+                }
+            }
+
+            stage('Delete the systems') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_systems")
+            }
+            stage('Delete config projects') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_config_projects")
+            }
+
+            stage('Delete software channels') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_software_channels")
+            }
+
+            stage('Delete activation keys') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_activation_keys")
+            }
+            stage('Delete minion users') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_users")
+            }
+            stage('Delete channel repositories') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_repositories")
+            }
+            stage('Delete salt keys') {
+                sh(script: "${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode delete_salt_keys")
+            }
```

Comment:
It looks very verbose, maybe we can have another string variable containing:  `"${SUSEManagerCleanerProgram} --url ${serverHostname} --product_version ${product_version} ${defaultResourcesToDeleteArgs} --mode "` ?